psutil/tests/__init__.py

```diff
@@ -380,20 +375,26 @@ def wait_for_pid(pid, timeout=GLOBAL_TIMEOUT):
             return
 
 
-def wait_for_file(fname, timeout=GLOBAL_TIMEOUT, delete_file=True):
-    """Wait for a file to be written on disk with some content."""
+def wait_for_file(fname, timeout=GLOBAL_TIMEOUT, empty=False,
+                  delete_file=True):
+    """Wait for a file to be written on disk with some content, or until
+    the file exists if empty=True."""
     stop_at = time.time() + timeout
+    sleep_for = 0.001
     while time.time() < stop_at:
         try:
             with open(fname, "r") as f:
                 data = f.read()
-            if not data:
+            if not empty and not data:
+                time.sleep(sleep_for)
+                sleep_for *= 2
                 continue
             if delete_file:
                 os.remove(fname)
             return data
-        except IOError:
-            time.sleep(0.001)
+        except (IOError, OSError):
+            time.sleep(sleep_for)
+            sleep_for *= 2
```

Comment:
This incremental timeout is a good change. I would put a limit though so that we sleep no more than, say, 0.01 secs per loop. That will make unit tests a bit faster.
`sleep_for *= 2` should become `sleep_for = min(sleep_for * 2, 0.01)`.
This way we'll wait:

```
0.001
0.002
0.004
0.008
0.01
0.01
...
```